tests/unit/test_charm.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+from unittest.mock import PropertyMock, patch
+
+from charm import SlurmdCharm
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+
+class TestCharm(unittest.TestCase):
```

Comment:
What is the purpose of using unittest classes? I think it would be preferable to use a generic python class and then implement a `setup()` method that is decorated by a pytest fixture with a "class" scope.

One nice side-effect is not having to have a method that uses camel-case for the method name.

Reply:
The standard for charm testing is using the python unit test framework with the Charmed Operator Framework testing harness, `Harness `. 

That being said, what you describe is similar to the scheme we use for integration tests though the pytest fixture is at a "module" scope.

Reply:
> python unit test framework with the Charmed Operator Framework testing harness

Will it work using `yield` in a fixture decorated method? I find it a little surprising to be using `unittest` classes here when pytest fixtures are used in other contexts within this testing suite.

Reply:
I think that could work, however, with `unittest` we can effectively test charm behavior without much complexity which fits with our requirements with `Harness`. There is also the benefit of keeping with charm unit test convention.